lib/Doctrine/ODM/PHPCR/DocumentManager.php

```diff
@@ -925,4 +925,16 @@ public function getNodeForDocument($document)
 
         return $this->session->getNode($path);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDocumentId($document)
+    {
+        try {
+            return $this->unitOfWork->getDocumentId($document);
+        } catch (PHPCRException $e) {
```

Comment:
getLocalesFor would be an example where we get an exception on an unmanaged document: https://github.com/doctrine/phpcr-odm/blob/master/lib/Doctrine/ODM/PHPCR/DocumentManager.php#L546 

`find` indeed does return null on not found. but i feel that is a design flaw of doctrine - i'd rather have an exception in that case. but i don't want to change that now, it would be confusing when orm does return null instead of ecxeption. however, new methods can be done in a more secure way.